graylog2-server/src/main/java/org/graylog/plugins/views/storage/migration/state/actions/MigrationActionsImpl.java

```diff
@@ -177,6 +180,13 @@ public boolean caAndRenewalPolicyExist() {
         return !caDoesNotExist() && !renewalPolicyDoesNotExist();
     }
 
+    @Override
+    public boolean compatibleDatanodesRunning() {
+        Map<String, DataNodeDto> nodes = nodeService.allActive();
+        return !nodes.isEmpty() && nodes.values().stream()
+                .allMatch(node -> Objects.equals(node.getDatanodeVersion(), graylogVersion.toString()));
```

Comment:
I think we should use compareTo method of the Version class itself, not fallbacking to String comparison. It may be a valid combination with different build information on each side and then the comparison would fail here.

Reply:
good point